File: Bridge/Kaliuzhnyi-03/ConsoleApp1/Shapes/Circle.cs
using Bridge.Colors;

namespace Bridge.Shapes;

public class Circle : Shape
{
    private readonly double _radius;
    public Circle(double radius, IColor color) : base(color)
    {
        if (radius <= 0)
        {
            throw new ArgumentException("Error! Radius can't be smaller or equal to zero!");
        }
        _radius = radius;
    }

    public override double Area()
    {
        return Math.Sqrt(_radius) * Math.PI;
    }

    public double Area(int roundPNumber)
    {
        if (roundPNumber < 0 || roundPNumber > 16)
        {
            throw new ArgumentException("Error! Rounding up is possible in range 0-16!");
        }
        return Math.Sqrt(_radius) * Math.Round(Math.PI, roundPNumber);
    }

    public override double Perimeter()
    {
        return 2 * _radius * Math.PI;
    }

    public double Perimeter(int roundPNumber)
    {
        if (roundPNumber < 0 || roundPNumber > 16)
        {
            throw new ArgumentException("Error! Rounding up is possible in range 0-16!");
        }
        return 2 * _radius * Math.Round(Math.PI, roundPNumber);
    }

    public override string ToString()
    {
        string circleToString = $"There is circle with radius = {_radius} and {Color} color";

        return circleToString;
    }
}

File: Bridge/Petukhov/src/lnu/edu/ua/shapes/Circle.java
package lnu.edu.ua.shapes;

import lnu.edu.ua.colors.Color;

public class Circle extends Shape {
    private double radius;

    public Circle(Color color, double radius) {
        super(color);
        this.radius = radius;
    }

    public Double getPerimeter() {
        return 2.0 * this.radius * Math.PI;
    }

    public Double getArea() {
        return Math.pow(this.radius, 2.0) * Math.PI;
    }

    @Override
    public String toString() {
        StringBuilder str = new StringBuilder("Circle:\n");
        str.append("radius: ").append(this.radius).append("\n");
        str.append("perimeter: ").append(getPerimeter()).append("\n");
        str.append("area: ").append(getArea()).append("\n");
        str.append("color: ").append(this.color.getColor()).append("\n");
        return str.toString();
    }
}